weaviate/connect/auth.py

```diff
@@ -0,0 +1,217 @@
+from __future__ import annotations
```

Comment:
I will remove `auth.py` for now and only use `authentication.py`, but we will need to use an async oidc-auth class when extending to async for both so that it's non-blocking. We can revisit that later